### PR TITLE
Rc june 2024

### PR DESCRIPTION
**Pull request by gonzalemario**

### CODE REVIEW

**Review by mikewallace1979 — APPROVED** [review]

File: pg_backup_api/news.yml [hunk 1]
```diff
@@ -1,8 +1,15 @@
-# © Copyright EnterpriseDB UK Limited 2021-2023 - All rights reserved.
+# © Copyright EnterpriseDB UK Limited 2021-2024 - All rights reserved.
 
 # pg-backup-api News - History of user-visible changes
 
 pg-backup-api release notes:
+  - version: "2.2.0"
+    date: "20240629"
+    changes:
+      Notable changes:
```
Comment (barthisrael) [inline pg_backup_api/news.yml:9]: I'd say to use bugfix instead of notable changes

**Review by barthisrael — CHANGES_REQUESTED** [review]
Besides the comment left in the changed files, as discussed in Slack I think we should just ignore Pyright errors for now (by setting `# pyright: ignore [reportCallIssue]` on both `exec_diagnose` lines) so we unblock the release for now.

Ideally that should've been implemented in a different way, so we wouldn't need to silence pyright. I mean, by using logic which depends on the Barman module version instead of on an `AttributeError` Exception. But changing that now would require some time to change the code, change the tests, perform tests again, ...

File: pg_backup_api/news.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
 # pg-backup-api release notes
 
-© Copyright EnterpriseDB UK Limited 2021-2023 - All rights reserved.
+© Copyright EnterpriseDB UK Limited 2021-2024 - All rights reserved.
+
+## 2.2.0 (2024-06-29)
+
+### Buxfixes
+
+- Functions/methods signatures updates to support Barman >= 3.9
```
Comment (barthisrael) [inline pg_backup_api/news.md:9]:
```suggestion
- Functions/methods signatures updates to fix regression with Barman <= 3.9
```

File: pg_backup_api/news.yml [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
+# © Copyright EnterpriseDB UK Limited 2021-2024 - All rights reserved.
 
 # pg-backup-api News - History of user-visible changes
 
 pg-backup-api release notes:
+  - version: "2.2.0"
+    date: "20240629"
+    changes:
+      Buxfixes:
+        - Functions/methods signatures updates to support Barman >= 3.9
```
Comment (barthisrael) [inline pg_backup_api/news.yml:10]:
```suggestion
        - Functions/methods signatures updates to fix regression with Barman <= 3.9
```

**Review by barthisrael — CHANGES_REQUESTED** [review]

File: pg_backup_api/version.txt [hunk 4]
```diff
@@ -1 +1 @@
-2.1.0
+2.2.0
```
Comment (barthisrael) [inline pg_backup_api/version.txt:1]: Shouldn't version be `2.2.1` in this case?
I mean, we are only shipping a bugfix.

**Review by barthisrael — CHANGES_REQUESTED** [review]

**Review by barthisrael — APPROVED** [review]

**Pull request merged**